library/common/include/c_types.h

```diff
@@ -61,14 +81,26 @@ typedef struct {
  */
 typedef struct {
   // Number of header elements in the array.
-  uint64_t length;
+  int length;
```

Comment:
Why is this no longer unsigned?

Reply:
We went back and forth.

Some convention still suggests `size_t` (though that's technically specifically for byte sizes, which this isn't, and also C++ committee apparently is mostly is on record saying they regret making it unsigned).

Google style guide recommends regular `int` in any case where an explicit bit width/sign behavior isn't actually required, and calls out that making a type unsigned offers no actual protection against people trying to set negative values.

Java uses signed lengths by convention.

Obj-c/swift use unsigned lengths by convention.

This was becoming a rabbit hole and `int` seemed like good neutral ground for now.

Reply:
Cool, was mostly just curious as to the thoughts behind the decision. Thanks for explaining!